Kitodo/src/main/java/de/sub/goobi/forms/AktuelleSchritteForm.java

```diff
@@ -182,38 +182,42 @@ public String takeOverBatch() {
         }
 
         for (Task task : currentStepsOfBatch) {
-            if (task.getProcessingStatusEnum().equals(TaskStatus.OPEN)) {
-                task.setProcessingStatusEnum(TaskStatus.INWORK);
-                task.setEditTypeEnum(TaskEditType.MANUAL_MULTI);
-                task.setProcessingTime(new Date());
-                User user = getUser();
-                serviceManager.getTaskService().replaceProcessingUser(task, user);
-                if (task.getProcessingBegin() == null) {
-                    task.setProcessingBegin(new Date());
-                }
+            processTask(task);
+        }
 
-                if (task.isTypeImagesRead() || task.isTypeImagesWrite()) {
-                    try {
-                        new File(serviceManager.getProcessService().getImagesOrigDirectory(false, task.getProcess()));
-                    } catch (Exception e1) {
-                        Helper.setErrorMessage("Error retrieving image directory: ", logger, e1);
-                    }
-                    task.setProcessingTime(new Date());
+        this.setBatchHelper(new BatchStepHelper(currentStepsOfBatch));
+        return "/pages/batchesEdit";
+    }
 
-                    serviceManager.getTaskService().replaceProcessingUser(task, user);
-                    this.myDav.downloadToHome(task.getProcess(), !task.isTypeImagesWrite());
-                }
+    private void processTask(Task task) {
+        if (task.getProcessingStatusEnum().equals(TaskStatus.OPEN)) {
+            task.setProcessingStatusEnum(TaskStatus.INWORK);
+            task.setEditTypeEnum(TaskEditType.MANUAL_MULTI);
+            task.setProcessingTime(new Date());
+            User user = getUser();
+            serviceManager.getTaskService().replaceProcessingUser(task, user);
+            if (task.getProcessingBegin() == null) {
+                task.setProcessingBegin(new Date());
             }
 
-            try {
-                this.serviceManager.getProcessService().save(task.getProcess());
-            } catch (DataException e) {
-                Helper.setErrorMessage("errorSaving", new Object[] {Helper.getTranslation("prozess") }, logger, e);
+            if (task.isTypeImagesRead() || task.isTypeImagesWrite()) {
+                try {
+                    new File(serviceManager.getProcessService().getImagesOrigDirectory(false, task.getProcess()));
```

Comment:
Even for moved code: Should here not an `exists()` call added or on a local variable an `exists()` call executed?  I would prefer last option.

Reply:
What do you mean?